Criterion/Legacy/Search/Gateway/CriterionHandler/PublishedOnDayOfWeek.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\CookbookBundle\Criterion\Legacy\Search\Gateway\CriterionHandler;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\Core\Persistence\Database\SelectQuery;
+use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter;
+use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriterionHandler;
+use EzSystems\CookbookBundle\Criterion as CustomCriterion;
+
+class PublishedOnDayOfWeek extends CriterionHandler
+{
+    const WEEKDAY_MAP = [
+        CustomCriterion\PublishedOnDayOfWeek::SUN => 1,
+        CustomCriterion\PublishedOnDayOfWeek::MON => 2,
+        CustomCriterion\PublishedOnDayOfWeek::TUE => 3,
+        CustomCriterion\PublishedOnDayOfWeek::WED => 4,
+        CustomCriterion\PublishedOnDayOfWeek::THU => 5,
+        CustomCriterion\PublishedOnDayOfWeek::FRI => 6,
+        CustomCriterion\PublishedOnDayOfWeek::SAT => 7,
+    ];
+    /**
+     * Check if this criterion handler accepts to handle the given criterion.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion $criterion
+     *
+     * @return bool
+     */
+    public function accept(Criterion $criterion)
+    {
+        return $criterion instanceof CustomCriterion\PublishedOnDayOfWeek;
+    }
+
+    /**
+     * @param \eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter $converter
+     * @param \eZ\Publish\Core\Persistence\Database\SelectQuery $query
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion $criterion
+     * @param array $languageSettings
+     * @return string
+     */
+    public function handle(
+        CriteriaConverter $converter,
+        SelectQuery $query,
+        Criterion $criterion,
+        array $languageSettings
+    ) {
+        $column = $this->dbHandler->quoteColumn('published', 'ezcontentobject');
+        $value = array_map([$this, 'mapCriterionValue'], $criterion->value);
+
+        return $query->expr->in(
+            // Note: MySQL-specific
+            "DAYOFWEEK(FROM_UNIXTIME($column))",
```

Comment:
So, we're saying that developer should create this gateway for specific DBMS he's using, right?

If that's the case maybe I should create gateways for SQLite and PostgreSQL? However I'm not sure how to inject proper one for [this test](https://github.com/alongosz/CookbookBundle/blob/741154a1d98382d7ebfbaedfafb51712a90d66b9/Tests/Criterion/PublishedOnDayOfWeekTest.php#L15). Of course I could skip it when the driver is not MySQL, but I feel there's a better way to do this.